### ISSUE TITLE
Using Swagger Api Spec to generate documentation

### ISSUE THREAD

**Issue by eznix86** [invertase/docs.page#319]
Implementation example: https://motif.land/guides/openapi
```json
{
...
api: { url: "swagger api endpont here" ,  doc: "can extend doc using mdx" }
...
}
```
It automatically adds API the side menu and extends the swagger doc with another mdx file.

**Comment by TranquilMarmot** [comment]
Yeah, this is what we use Mintlify for. It's very similar to docs.page but is _very_ expensive.

**Comment by Ehesp** [comment]
I'll add this one to track. There's a big update in the works so I'll look into this once that lands.

Question: does it make sense to source the open api spec from a remote url or as a static file via github? (or both?)

**Comment by alexdukeinvertase** [comment]
As disscused @Ehesp , likely direction is OpenAPI-generated API reference as part of our next roadmap iteration.

**Comment by Ehesp** [comment]
So I did some investigation into this... 

Overall it's possible (remote `$ref` support would not be), however some thoughts:

- We'd need to keep it "on demand" rendering to align with docs.page
- Some specs (e.g. Stripe, GitHub) are huge, in the realms of MBs.
- Obviously parsing this on every request isn't really feasible. Caching helps, but doesn't remove the overhead of parsing.
- So generally we'd need to come up with a clever caching strategy which breaks down a spec into what drives docs.page; URL -> identifier, sidebar titles -> URLs, creating a cached spec-per-page setup, so we essentially cache what's already been parsed (kinda like we do for github docs).

Needs a lot more thought, but interesting concept.
